draft-ietf-tls-esni.md

```diff
@@ -455,21 +497,34 @@ Finally, the client MUST generate a ClientHelloOuter message containing the
 "encrypted_client_hello" extension with the values as indicated above. In
 particular,
 
-- suite contains the client's chosen HpkeCipherSuite;
-- record_digest contains the digest of the corresponding ECHConfig structure;
+- cipher_suite contains the client's chosen HpkeCipherSuite;
+- config_digest contains the digest of the corresponding ECHConfig structure;
 - enc contains the encapsulated key as output by SetupBaseS; and
 - encrypted_ch contains the HPKE encapsulated key (enc) and the ClientHelloInner
   ciphertext (encrypted_ch_inner).
 
-The client MUST place the value of ECHConfig.public_name in the ClientHelloOuter
-"server_name" extension. The ClientHelloOuter MUST NOT contain a "cached_info"
-extension {{!RFC7924}} with a CachedObject entry whose CachedInformationType is
-"cert", since this indication would divulge the true server name. The remaining
-contents of the ClientHelloOuter MAY be identical to those in ClientHelloInner
-but MAY also differ.
+[[TODO: This parameter list has already been defined. I suggest cutting it
+from here.]]
+
+The client MUST place the value of `ECHConfig.public_name` in the
+ClientHelloOuter "server_name" extension. [[TODO: What does this mean? "The
+client MUST place the value of `ECHConfig.public_name` with the value of
+'server_name' extension"?]] The ClientHelloOuter MUST NOT contain a
+"cached_info" extension {{!RFC7924}} with a CachedObject entry whose
+CachedInformationType is "cert", since this indication would divulge the true
+server name.
+[[TODO: It seems like this should be worked out more in detail in security
+considerations. There might be other (future) extensions that interact with ECH
+in a way this document can't control.]]
+The remaining contents of the ClientHelloOuter MAY be identical to those in
+ClientHelloInner but MAY also differ.
 
 ## Recommended Padding Scheme {#padding}
 
+[[TODO: I don't understand the purpose of the padding scheme. Is this meant to
```

Comment:
It's a RECOMMENDED padding scheme. What's the confusion?

Reply:
I'll resolve this TOOD for now.